### PR TITLE
Uat/86cwh987k artist availability

### PR DESCRIPTION
**Pull request by scp777**
**Ticket:**
https://app.clickup.com/t/86cwh987k

Added availability tab to user profile page.

**Instructions and Results:**

1. Create a new artist account
2. User should see a red dot next to the newly created "availability" tab.
3. Make an availability entry for the current week.
4. Red dot should go away.

Screenshot 2024-10-01 at 12 46 00 PM

Screenshot 2024-10-01 at 12 46 09 PM

### CODE REVIEW

**Comment by joshualoongwy** [comment]
Some unresolved merge conflicts atm

**Comment by scp777** [comment]
conflicts resolved :)

File: behind-the-veil-siteroot/imports/ui/components/pages/profile/ArtistProfileTabs.jsx [hunk 1]
```diff
@@ -24,49 +26,94 @@ import { useUserBookings } from "../../DatabaseHelper.jsx";
...
+
+      setWeeklyAvailabilityEntered(hasDateInCurrentWeek);
+    };
+
+    if (userData?.availability) {
+      checkForCurrentWeekAvailability(userData.availability);
+    }
+  }, [userData]);
+
+  console.log(weeklyAvailabilityEntered);
```
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/pages/profile/ArtistProfileTabs.jsx:62]: leftover log

File: behind-the-veil-siteroot/imports/ui/components/pages/request-booking/RequestBooking.jsx
Comment (joshualoongwy) [inline behind-the-veil-siteroot/imports/ui/components/pages/request-booking/RequestBooking.jsx:1]: Looks like this is mostly indentation changes, do you remember if there were any functional code changes that need to be reviewed in this file, cos my brain is having difficulty comprehending due to the size

Comment (scp777) [reply behind-the-veil-siteroot/imports/ui/components/pages/request-booking/RequestBooking.jsx:1]: Yes, I was a little confused as well as I didn't remember adding code. I think I saved it and it triggered Prettier to make changes to the indentation and formatting. But no logic change.

File: behind-the-veil-siteroot/imports/ui/components/pages/profile/ArtistProfileTabs.jsx (hunk 1 above)
Comment (scp777) [reply behind-the-veil-siteroot/imports/ui/components/pages/profile/ArtistProfileTabs.jsx]: Deleted and pushed new commit

File: behind-the-veil-siteroot/imports/ui/components/pages/add-availability/AddAvailability.jsx
[the remark this answers is not in the document]
Comment (scp777) [reply behind-the-veil-siteroot/imports/ui/components/pages/add-availability/AddAvailability.jsx:1]: This was something I wasn't entirely sure about and might need to ask the team. Personally, I think if there's already a dedicated tab for "Add Availability", then we can reduce clutter and get rid of AddAvailability.jsx. 

But yes, at the moment, the only difference is the styling to accommodate the tab component.

**Review by joshualoongwy — APPROVED** [review]

**Pull request merged**